### PR TITLE
echo: Some minor changes to options

### PR DESCRIPTION
**Pull request by alstolten**
Changed from static to const.
Added alternative usage, see GNU echo.
Do not allow short options for --version and --help, compare also to GNU echo.
Added tests for the above.

### CODE REVIEW

**Comment by sylvestre** [comment]
>  Do not allow short options for --version and --help, compare also to GNU echo.

Why do we want to do that?
This makes the code more complex

**Comment by alstolten** [comment]
> > Do not allow short options for --version and --help, compare also to GNU echo.
> 
> Why do we want to do that?
> This makes the code more complex

I thought that it might be usefull to mimic the exact behavior, e.g. to not break scripts etc.
But if you do not see fit thats also fine.

**Comment by sylvestre** [comment]
Mimicking the behavior is one of the goal but not for every cases. For example, here, not sure to see the value :)

Could you please remove this part of the PR ? thanks

File: src/uu/echo/src/echo.rs [hunk 1]
```diff
@@ -123,7 +120,6 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         .setting(clap::AppSettings::TrailingVarArg)
         .setting(clap::AppSettings::AllowLeadingHyphen)
         .version(crate_version!())
-        .usage(USAGE)
```
Comment (sylvestre) [inline src/uu/echo/src/echo.rs:126]: why did you remove it?
it was useful
we could use get_usage like the rest of the program,

File: src/uu/echo/src/echo.rs [hunk 2]
```diff
@@ -149,7 +145,6 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         )
         .arg(
             Arg::with_name(options::STRING)
-                .hidden(true)
```
Comment (ycd) [inline src/uu/echo/src/echo.rs:152]: This should stay as hidden

**Review by ycd — CHANGES_REQUESTED** [review]

File: src/uu/echo/src/echo.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/echo/src/echo.rs:126]: ping?

Comment (alstolten) [reply src/uu/echo/src/echo.rs:126]: :)
I also value consistency. I changed it back. 

However for me it still makes no sense, as you need to write a lot of extra code with no added value (in the ls example the function get_usage() is only used at exactly one spot, where the built-in pattern of clap could be leveraged, and instead you need to maintain a string and a fn. But maybe I miss something here.

**Comment by sylvestre** [comment]
thanks!
